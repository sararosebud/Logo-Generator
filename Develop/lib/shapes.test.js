const { Circle, Triangle, Square } = require('./shapes');

describe('Shapes', () => {
  describe('Circle', () => {
    it('should render correctly', () => {
      const circle = new Circle();
      circle.setColor('#ff0000');
      circle.setText('ABC', '#ffffff');
      const svg = circle.render();
      expect(svg).toMatchSnapshot();
    });
  });

  describe('Triangle', () => {
    it('should render correctly', () => {
      const triangle = new Triangle();
      triangle.setColor('#00ff00');
      triangle.setText('XYZ', '#000000');
      const svg = triangle.render();
      expect(svg).toMatchSnapshot();
    });
  });

  describe('Square', () => {
    it('should render correctly', () => {
      const square = new Square();
      square.setColor('#0000ff');
      square.setText('123', '#ffffff');
      const svg = square.render();
      expect(svg).toMatchSnapshot();
    });
  });
});
