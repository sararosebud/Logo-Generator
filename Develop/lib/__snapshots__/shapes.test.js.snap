// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapes Circle should render correctly 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" 
        width="200" height="300">
          <g>
            <circle cx="100" cy="100" r="90" fill ="#ff0000">
            </circle>
            <text x="50%" y="35%" text-anchor="middle" font-size="60" fill="#ffffff" dy=".3em">ABC</text>
          </g>
        </svg>"
`;

exports[`Shapes Square should render correctly 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" 
        width="300" height="200">
        <g transform="translate(50,10)">
        <rect width="200" height="200" fill="#0000ff"></rect>
        <text x="100" y="100" alignment-baseline="middle" font-size="55" fill="#ffffff" text-anchor="middle">123</text>
      </g>    
        </svg>"
`;

exports[`Shapes Triangle should render correctly 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" 
        width="300" height="200">
          <g>
          <polygon points="150, 18 244, 182 56, 182" fill="#00ff00">
          </polygon>
          <text x="150" y="125" text-anchor="middle" font-size="56" fill="#000000" dy=".3em">XYZ</text>
          </g>
          </svg>"
`;
